pyro/params/param_store.py

```diff
@@ -120,6 +123,21 @@ def get_param(self, name, init_tensor=None, constraint=constraints.real):
 
         return param
 
+    def get_params(self, name, constraint=constraints.real):
```

Comment:
1. Consider renaming this to `.match()` or `.search()` or `.params_matching()`
2. I don't think you want a `constraint` kwarg; it is only useful for creating params in the param store (when combined with an `init_tensor`). I believe it will always be ignored here, since you're only iterating over params that already exist.

Reply:
1. sure, though i wanted the name to denote returning multiple params
2. ah right the `_constraints` registry already has all the constrained values at call time